Limit animations to specific properties

ConfusedVorlon opened the pull request:
as things stand, buttons sometimes animate to their starting positions because the button style is providing an animation which is then used by the system for the position.

see this video for an example:
https://www.dropbox.com/s/6am1lkcz79g6xzv/current_animation.mov?dl=0

when the sheet appears - the button animates to the centre spot.

specifying the animation value in elevation.swift fixes this
(essentially telling swiftUI that this animation only applies to changes of elevation

see fixed version:
https://www.dropbox.com/s/gq8ni7ktsm82c2a/fixed_animation.mov?dl=0

I have added specific values everywhere there is an animation as that seems like a good practice to keep the animations encapsulated. I haven't tested on tvos. Also - it's not my code, so please do review and check that I have correctly identified the correct single value which is changing to drive the animations.

chkn commented on Sources/MaterialUI/DisabledFader.swift:
```diff
@@ -15,6 +15,6 @@ struct DisabledFader: ViewModifier {
 			.compositingGroup()
 			.opacity(isEnabled ? 1 : 0.5)
-			.animation(nil)
+            .animation(nil,value:isEnabled)
```
```suggestion
			.animation(nil, value: isEnabled)
```

chkn commented on Sources/MaterialUI/Elevation.swift:
```diff
@@ -53,15 +53,15 @@ fileprivate struct Elevation: ViewModifier {
 	{
 		content
-			.animation(nil)
+            .animation(nil,value:elevation)
```
```suggestion
			.animation(nil, value: elevation)
```

chkn commented on Sources/MaterialUI/Elevation.swift:
```diff
@@ -53,15 +53,15 @@ fileprivate struct Elevation: ViewModifier {
 
 			// FIXME: Is this the correct curve?
-			.animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil)
+            .animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil,value:elevation)
```
```suggestion
			.animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil, value: elevation)
```

chkn commented on Sources/MaterialUI/Ripple.swift:
```diff
@@ -60,7 +60,7 @@ fileprivate struct Ripple: ViewModifier {
 			.offset(rippleOffset(mX, mY))
 			.scaleEffect(rippleAppeared ? 1 : Self.kRippleStartingScale)
-			.animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration))
+            .animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration),value:rippleAppeared)
```
```suggestion
			.animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration), value: rippleAppeared)
```

chkn commented on Sources/MaterialUI/Ripple.swift:
```diff
@@ -73,14 +73,14 @@ fileprivate struct Ripple: ViewModifier {
 							.fill(color.opacity(pointerState.isOver ? 0.08 : 0))
 							.frame(width: dimension, height: dimension)
-							.animation(.linear(duration: Self.kRippleFadeInOutDuration))
+                            .animation(.linear(duration: Self.kRippleFadeInOutDuration),value:pointerState.isOver)
```
```suggestion
							.animation(.linear(duration: Self.kRippleFadeInOutDuration), value: pointerState.isOver)
```

chkn commented on Sources/MaterialUI/Ripple.swift:
```diff
@@ -73,14 +73,14 @@ fileprivate struct Ripple: ViewModifier {
 							.fill(color.opacity(0.16))
 							.frame(width: dimension, height: dimension)
-							.animation(.linear(duration: Self.kRippleFadeInOutDuration))
+							.animation(.linear(duration: Self.kRippleFadeInOutDuration),value:pointerState.isOver)
```
```suggestion
							.animation(.linear(duration: Self.kRippleFadeInOutDuration), value: pointerState.isOver)
```

chkn commented on Sources/MaterialUI/Toggle.swift:
```diff
@@ -41,12 +41,12 @@ public struct MaterialSwitchToggleStyle: ToggleStyle {
 				.frame(width: Self.thumbDiameter, height: Self.thumbDiameter)
 				.offset(offset)
-				.animation(anim)
+                .animation(anim,value:configuration.isOn)
```
```suggestion
				.animation(anim, value: configuration.isOn)
```

chkn commented on Sources/MaterialUI/Toggle.swift:
```diff
@@ -41,12 +41,12 @@ public struct MaterialSwitchToggleStyle: ToggleStyle {
 			.modifier(DisabledFader())
 			.radialRipple(configuration.isOn ? Color.accentColor : Color.primary, offset: offset)
-			.animation(anim)
+			.animation(anim,value:configuration.isOn)
```
```suggestion
			.animation(anim, value: configuration.isOn)
```

chkn reviewed (commented):
Good catch finding that animation bug! I think you got all the right values- at least, all the animations still appear correct to me. Just a few code style nitpick issues and then we can merge this.

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon commented:
I think this is the first time I have actually hit tabs vs spaces in real life :)
also - confirming MIT licence is fine.

chkn reviewed (approved):
This looks good, thanks for making those changes! Looks like it just has some conflicts now after merging your other PR. Once those are fixed this can be merged

ConfusedVorlon committed:
Merge branch 'master' into feature/animation_fix

ConfusedVorlon commented:
updated

chkn commented:
And merged! Thanks again!

ConfusedVorlon commented:
Thanks for being so quick and involved. It's refreshing when pull requests are not ignored!

ConfusedVorlon committed:
Limit animations to specific properties

doing this in elevation.swift fixes the layout bug I have found (where the position of a button animates on appear)

I have added this everywhere there is an animation as that seems like a good practice to keep the animations encapsulated
